src/ucp/proto/proto_common.inl

```diff
@@ -192,24 +200,32 @@ out_put_request:
 static UCS_F_ALWAYS_INLINE size_t
 ucp_proto_request_pack_rkey(ucp_request_t *req, void *rkey_buffer)
 {
+    ucs_memory_info_t mem_info = {
```

Comment:
all of UCP code should use ucp_memory_info_t, except when using memtype cache

Reply:
@yosefe We need to get base address and length here https://github.com/openucx/ucx/pull/6739/files#diff-9136600ae0324c21c07ccead535b761ee5a5113c2817d02af656b33ce992ba8dR123 for whole allocation registration. For this, we would need to pass ucs_memory_info_t. This forces use of ucs_memory_info_t instead of ucp_memory_info_t for functions such as ucp_memory_info_set_host() and in turn functions which call info_set_host(). Any thoughts on how to address this?

Reply:
@yosefe I've used ucp_memory_info_t in most places now. Can you lmk if the changes look good?